client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/GlobalScopedPrivilege.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security.user.privileges;
+
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents generic global application privileges that can be scoped for each
+ * application. The privilege definition, as well as the scope identifier, are
+ * outside of the Elasticsearch jurisdiction.
+ */
+public class GlobalScopedPrivilege {
+
+    private final String scope;
+    private final Map<String, Object> privilege;
```

Comment:
I wonder if it would help to include the "category" (`"application"`, and in theory `"settings"`) in this class too.
We can wait add it in the future if we need it, but I think we will, depending on how `GlobalPrivileges` ends up looking.

Reply:
Hmmm, I cannot think of a reason why not to do that.
I am now leaning towards having a set in `GlobalPrivileges` because a map would be exposing unnecessary implementation details.

Reply:
This was a very good idea. I know this is how they are modeled on the server, but I didn't give it much interest at that time.